### PR TITLE
Adds query string query documentation

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Adds query string query documentation

### Checklist
- [x ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _opensearch/query-dsl/query-string.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    "query_string": {
+      "query": "speaker:KING  AND play_name: *well"
+    }
+  }
+}
+```
+
+## Parameters
+
+Query string query supports the following parameters. All parameters except `query` are optional.
```
Comment (vagimeli) [inline _opensearch/query-dsl/query-string.md:29]: Should this follow the same format as the first sentence, that is, "`query string` query?"

File: _opensearch/query-dsl/query-string.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+`default_field` | String | The field in which to search if the field is not specified in the query string. Supports wildcards. Defaults to the value specified in the `index.query.default_field` index setting. By default, the `index.query.default_field` is `*`, which means extract all fields eligible for term query and filter the metadata fields. The extracted fields are combined into a query if the `prefix` is not specified. Eligible fields do not include nested documents. Searching all eligible fields could be a resource-intensive operation. The `indices.query.bool.max_clause_count` search setting defines the maximum value for the product of the number of fields and the number of terms that can be queried at one time. The default value for `indices.query.bool.max_clause_count` is 4,096.
+`allow_leading_wildcard` | Boolean | Specifies whether `*` and `?` are allowed as first characters of a search term. Default is `true`.
+`analyze_wildcard` | Boolean | Specifies whether OpenSearch should attempt to analyze wildcard terms. Some analyzers do a poor job at this task, so the default is `false`.
```
Comment (vagimeli) [inline _opensearch/query-dsl/query-string.md:37]: Are we saying that some of OpenSearch analyzers do a poor job? Do we need to qualify why the default is false? Suggestion: Delete the qualifier so the sentence reads "The default is `false`."

File: _opensearch/query-dsl/query-string.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+`rewrite` | String | Determines how OpenSearch rewrites and scores multi-term queries. Valid values are `constant_score`, `scoring_boolean`, `constant_score_boolean`, `top_terms_N`, `top_terms_boost_N`, and `top_terms_blended_freqs_N`. Default is `constant_score`.
+`auto_generate_synonyms_phrase_query` | Boolean | Specifies whether to create [match queries]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/full-text#match) automatically for multi-term synonyms. Default is `true`.
+`boost` | Floating-point | Boosts the clause by the given multiplier. Values less than 1.0 decrease relevance, and values greater than 1.0 increase relevance. Default is 1.0. 
+`default_operator`| String | The default Boolean operator used if no operators are specified. Valid values are:<br>- `OR`: The string `to be` is interpreted as `to OR be`<br>- `AND`: The string `to be` is interpreted as `to AND be`<br> Default is `OR`.
```
Comment (vagimeli) [inline _opensearch/query-dsl/query-string.md:46]: Insert space after "are" and before ":"

**Review by vagimeli — APPROVED** [review]
Great writing. Minimal comments. LGTM

File: _opensearch/query-dsl/query-string.md (hunk 2 above)
Comment (kolchfa-aws) [reply _opensearch/query-dsl/query-string.md:37]: Could not find any justification as to why the default is false, so removing the justification.

File: _opensearch/query-dsl/query-string.md (hunk 3 above)
Comment (kolchfa-aws) [reply _opensearch/query-dsl/query-string.md:46]: The `<br>` puts the list on a different line, so no space needed.

**Review by hdhalter — COMMENTED** [review]
Do we say anywhere why they would do this? For example, "You can search your rollup indexes using query string search queries." Or this, taken from the release blog: "OpenSearch users use index rollups to compress older time-series data into summarized indexes, reducing data granularity and offering potential improvements in storage costs and performance. You can use query string queries to search your rollup indexes and find the data you’re looking for."

**Review by Naarcha-AWS — APPROVED** [review]

**Comment by kolchfa-aws** [comment]
This PR will be backported to previous versions, so I won't mention rollups here since it's only 2.5 and above. We can add when we enhance this doc in a separate PR

**Pull request merged**